<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>HttpListenerConfigDlg</class>
 <widget class="QDialog" name="HttpListenerConfigDlg">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>902</width>
    <height>725</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <widget class="control::MwcLabelLarge" name="titleLabel">
   <property name="geometry">
    <rect>
     <x>295</x>
     <y>15</y>
     <width>313</width>
     <height>40</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>40</height>
    </size>
   </property>
   <property name="text">
    <string>FOREIGN API LISTENER</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget3">
   <property name="geometry">
    <rect>
     <x>120</x>
     <y>660</y>
     <width>676</width>
     <height>40</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="spacing">
     <number>30</number>
    </property>
    <item>
     <widget class="control::MwcPushButtonNormal" name="resetButton">
      <property name="minimumSize">
       <size>
        <width>150</width>
        <height>40</height>
       </size>
      </property>
      <property name="focusPolicy">
       <enum>Qt::StrongFocus</enum>
      </property>
      <property name="text">
       <string>Reset</string>
      </property>
     </widget>
    </item>
    <item>
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="control::MwcPushButtonNormal" name="cancelButton">
      <property name="minimumSize">
       <size>
        <width>150</width>
        <height>40</height>
       </size>
      </property>
      <property name="focusPolicy">
       <enum>Qt::StrongFocus</enum>
      </property>
      <property name="text">
       <string>Cancel</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="control::MwcPushButtonNormal" name="applyButton">
      <property name="minimumSize">
       <size>
        <width>150</width>
        <height>40</height>
       </size>
      </property>
      <property name="focusPolicy">
       <enum>Qt::StrongFocus</enum>
      </property>
      <property name="text">
       <string>Apply</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="layoutWidget1_2">
   <property name="geometry">
    <rect>
     <x>365</x>
     <y>225</y>
     <width>486</width>
     <height>63</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <property name="spacing">
     <number>5</number>
    </property>
    <item>
     <widget class="control::MwcLabelSmall" name="listeningAddress_label">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>22</height>
       </size>
      </property>
      <property name="text">
       <string>Listening Address: ip:port</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="control::MwcLineEditNormal" name="listeningAddressEdit">
      <property name="minimumSize">
       <size>
        <width>250</width>
        <height>40</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>20000</width>
        <height>40</height>
       </size>
      </property>
      <property name="maxLength">
       <number>128</number>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QCheckBox" name="useBasicAutorization">
   <property name="geometry">
    <rect>
     <x>45</x>
     <y>340</y>
     <width>296</width>
     <height>26</height>
    </rect>
   </property>
   <property name="text">
    <string>Use basic authorization</string>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget1_5">
   <property name="geometry">
    <rect>
     <x>365</x>
     <y>315</y>
     <width>311</width>
     <height>63</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_6">
    <property name="spacing">
     <number>5</number>
    </property>
    <item>
     <widget class="control::MwcLabelSmall" name="apiSecret_label">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>22</height>
       </size>
      </property>
      <property name="text">
       <string>Password for user 'mwc'</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="control::MwcLineEditNormal" name="apiSecretEdit">
      <property name="minimumSize">
       <size>
        <width>250</width>
        <height>40</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>20000</width>
        <height>40</height>
       </size>
      </property>
      <property name="maxLength">
       <number>128</number>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QCheckBox" name="useTlsCheck">
   <property name="geometry">
    <rect>
     <x>45</x>
     <y>410</y>
     <width>446</width>
     <height>26</height>
    </rect>
   </property>
   <property name="text">
    <string>Use Secure HTTPS connection</string>
   </property>
  </widget>
  <widget class="control::MwcLabelNormal" name="titleLabel_2">
   <property name="geometry">
    <rect>
     <x>15</x>
     <y>65</y>
     <width>871</width>
     <height>146</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>40</height>
    </size>
   </property>
   <property name="text">
    <string>WARNING!!!  HTTP(S) configuration requires advanced technical level. Normally it is not enough just to enter configuration parameters. More complicated configuration need to be done on network and OS level.

For your security MWC team recommends to obtain certificates and use https connection with basic authorization.

Note, improper HTTP configuration can make your computer vulnerable.</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
   <property name="wordWrap">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QCheckBox" name="activateRestApi">
   <property name="geometry">
    <rect>
     <x>45</x>
     <y>250</y>
     <width>296</width>
     <height>26</height>
    </rect>
   </property>
   <property name="text">
    <string>Activate Foreign REST API listener</string>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget1_6">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>455</y>
     <width>626</width>
     <height>63</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_7">
    <property name="spacing">
     <number>5</number>
    </property>
    <item>
     <widget class="control::MwcLabelSmall" name="tlsPrivateKey_label">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>22</height>
       </size>
      </property>
      <property name="text">
       <string>TLS certificate private key, pem file</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="control::MwcLineEditNormal" name="tlsPrivateKeyEdit">
      <property name="minimumSize">
       <size>
        <width>250</width>
        <height>40</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>20000</width>
        <height>40</height>
       </size>
      </property>
      <property name="maxLength">
       <number>128</number>
      </property>
      <property name="placeholderText">
       <string>path to certificate private key file</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="control::MwcPushButtonNormal" name="selectPrivKeyButton">
   <property name="geometry">
    <rect>
     <x>705</x>
     <y>478</y>
     <width>150</width>
     <height>40</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>150</width>
     <height>40</height>
    </size>
   </property>
   <property name="focusPolicy">
    <enum>Qt::StrongFocus</enum>
   </property>
   <property name="text">
    <string>Select File</string>
   </property>
  </widget>
  <widget class="control::MwcPushButtonNormal" name="selectFullchainButton">
   <property name="geometry">
    <rect>
     <x>705</x>
     <y>563</y>
     <width>150</width>
     <height>40</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>150</width>
     <height>40</height>
    </size>
   </property>
   <property name="focusPolicy">
    <enum>Qt::StrongFocus</enum>
   </property>
   <property name="text">
    <string>Select File</string>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget1_7">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>540</y>
     <width>626</width>
     <height>63</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_8">
    <property name="spacing">
     <number>5</number>
    </property>
    <item>
     <widget class="control::MwcLabelSmall" name="tlsFullchain_label">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>22</height>
       </size>
      </property>
      <property name="text">
       <string>TLS fullchain certificate, pem file</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="control::MwcLineEditNormal" name="tlsFullchainEdit">
      <property name="minimumSize">
       <size>
        <width>250</width>
        <height>40</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>20000</width>
        <height>40</height>
       </size>
      </property>
      <property name="maxLength">
       <number>128</number>
      </property>
      <property name="placeholderText">
       <string>path to fullchain certificate</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="control::MwcPushButtonNormal" name="generateSecretButton">
   <property name="geometry">
    <rect>
     <x>705</x>
     <y>338</y>
     <width>150</width>
     <height>40</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>150</width>
     <height>40</height>
    </size>
   </property>
   <property name="focusPolicy">
    <enum>Qt::StrongFocus</enum>
   </property>
   <property name="text">
    <string>Generate</string>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>control::MwcPushButtonNormal</class>
   <extends>QPushButton</extends>
   <header>control/MwcPushButton.h</header>
  </customwidget>
  <customwidget>
   <class>control::MwcLineEditNormal</class>
   <extends>QLineEdit</extends>
   <header>control/MwcLineEdit.h</header>
  </customwidget>
  <customwidget>
   <class>control::MwcLabelSmall</class>
   <extends>QLabel</extends>
   <header>control/MwcLabel.h</header>
  </customwidget>
  <customwidget>
   <class>control::MwcLabelNormal</class>
   <extends>QLabel</extends>
   <header>control/MwcLabel.h</header>
  </customwidget>
  <customwidget>
   <class>control::MwcLabelLarge</class>
   <extends>QLabel</extends>
   <header>control/MwcLabel.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>activateRestApi</tabstop>
  <tabstop>listeningAddressEdit</tabstop>
  <tabstop>useBasicAutorization</tabstop>
  <tabstop>apiSecretEdit</tabstop>
  <tabstop>generateSecretButton</tabstop>
  <tabstop>useTlsCheck</tabstop>
  <tabstop>tlsPrivateKeyEdit</tabstop>
  <tabstop>selectPrivKeyButton</tabstop>
  <tabstop>tlsFullchainEdit</tabstop>
  <tabstop>selectFullchainButton</tabstop>
  <tabstop>resetButton</tabstop>
  <tabstop>cancelButton</tabstop>
  <tabstop>applyButton</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
